lib/logstash/outputs/opsgenie.rb

```diff
@@ -6,6 +6,27 @@
 require "net/http"
 require "net/https"
 
+
+HTTP_EXCEPTIONS = [
+  Errno::EINVAL,
+  Net::HTTPBadResponse,
+  Net::HTTPHeaderSyntaxError,
+]
+
+RETRYABLE_HTTP_EXCEPTIONS = [
```

Comment:
All those are just different stages of a network connection failing. For us in production we have seen SSLError and EOFError but I bet the other can also happen.